server/src/main/java/org/elasticsearch/index/mapper/RootObjectMapper.java

```diff
@@ -326,7 +326,13 @@ protected void doMerge(ObjectMapper mergeWith, MergeReason reason) {
             }
         }
         assert this.runtimeFieldTypes != mergeWithObject.runtimeFieldTypes;
-        this.runtimeFieldTypes.putAll(mergeWithObject.runtimeFieldTypes);
+        for (Map.Entry<String, RuntimeFieldType> runtimeField : mergeWithObject.runtimeFieldTypes.entrySet()) {
+            if (runtimeField.getValue().isPlaceholderForRemoval()) {
```

Comment:
Possibly dumb question: is there any reason we can't just use `null` as the placeholder here?

Reply:
This is all but a dumb question! This used to be necessary because we'd build a DocumentMapper, which includes not only Mapping but also MappingLookup and FieldTypeLookup, and they could not really support a null type. At this point though, I may have made that unnecessary without realizing. Let me give this a try.

Reply:
Unfortunately the placeholder is still needed. Once we move away from constructing DocumentMapper before merging, and we will remove DocumentMapper#merge, we will be able to easily remove the placeholder. I will note this down.

Reply:
actually, I looked even more closely, and it seems pretty easy to address this, I will open a separate PR for it.

Reply:
I have removed the placeholder, it is no longer needed after merging #69005 in.